### ISSUE TITLE
Missing expression after unary operator '--'

### ISSUE THREAD

**Issue by cdsre** [riemers/ansible-gitlab-runner#268]
Deploying windows gitlab runner using the latest v2.0.0 results in the following error

```json
{
  "changed": true,
  "cmd": "c:/gitlab-runner//gitlab-runner.exe register --non-interactive --url 'https://gitlab.example.com' --description 'Windows Docker Runners - MY-SERVER-NAME' --tag-list 'docker-windows'    --executor 'docker-windows'  --limit '0' --output-limit '10240' --locked='False' --docker-image 'mcr.microsoft.com/windows/servercore:1809' --docker-volumes \"\\\\.\\pipe\\docker_engine:\\\\.\\pipe\\docker_engine\" \n --registration-token 'my-secret-registration-token'",
  "delta": "0:00:02.094302",
  "end": "2023-06-25 10:17:01.020912",
  "msg": "non-zero return code",
  "rc": 1,
  "start": "2023-06-25 10:16:58.926610",
  "stderr": "At line:2 char:4\r\n+ --registration-token'my-secret-registration-token'\r\n+    ~\r\nMissingexpression after unary operator '--'.\r\nAt line:2 char:4\r\n+  --registration-token 'my-secret-registration-token'\r\n+    ~~~~~~~~~~~~~~~~~~\r\nUnexpected token 'registration-token' in expression or statement.\r\n    + CategoryInfo : ParserError: (:) [], ParentContainsErrorRecordException\r\n    + FullyQualifiedErrorId : MissingExpressionAfterOperator",
  "stderr_lines": [
    "At line:2 char:4",
    "+  --registration-token 'my-secret-registration-token'",
    "+ ~",
    "Missing expression after unary operator '--'.",
    "At line:2 char:4",
    "+ --registration-token 'my-secret-registration-token'",
    "+   ~~~~~~~~~~~~~~~~~~",
    "Unexpected token 'registration-token' in expression or statement.",
    "    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException",
    "    + FullyQualifiedErrorId : MissingExpressionAfterOperator"
  ],
  "stdout": "",
  "stdout_lines": []
}
```

The key point to note here is the new line char before the `--registration-token` flag. This essentially breaks this into two seperate commands where the second command is ` --registration-token 'my-secret-registration-token'` and powershell errors thinking the double hyphen is a unary operator to decrease a value. 

This can be simulated in powershell using a new line char before the token

```powershell
 echo "foo"                                                                              12:55:07 
>  --registration 'foo'
At line:2 char:4
+  --registration 'foo'
+    ~
Missing expression after unary operator '--'.
At line:2 char:4
+  --registration 'foo'
+    ~~~~~~~~~~~~
Unexpected token 'registration' in expression or statement.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : MissingExpressionAfterOperator
```

The issue here is the change to use set_fact to build the runner command where previourly this was done all in the same block as registering the runner token. While the yaml is using the folded style to replace the new lines with spaces. I suspect the way ansible and jinja templating works is the yaml folding will happen before templating so 

```yaml
- name: (Windows) Register runner to GitLab
  win_shell: >
    {{ register_runner_cmd }}
    --registration-token '{{ gitlab_runner.token|default(gitlab_runner_registration_token) }}'
    {% if gitlab_runner.cache_s3_secret_key is defined %}
    --cache-s3-secret-key '{{ gitlab_runner.cache_s3_secret_key }}'
    {% endif %}
    {% if gitlab_runner.ssh_password is defined %}
    --ssh-password '{{ gitlab_runner.ssh_password }}'
    {% endif %}
```

will be folded into 

```
 {{ register_runner_cmd }} --registration-token '{{ gitlab_runner.token|default(gitlab_runner_registration_token) }}'
```

Then the values will be interpolated. but at this point the register_runner_cmd will have a new line inside it which causes the issue here. A small demo of this with ansible can be shown like this

```yaml
- name: testing
  hosts: localhost
  gather_facts: false
  tasks:
    - set_fact:
        my_cmd: >
          'hello'
          'world'
    - debug:
        var: my_cmd
    - shell:
        cmd: >
          echo
          {{ my_cmd }}
          'foobar'
```

This produces the output 

```shell
TASK [set_fact] ****************************************************************
Monday 26 June 2023  13:15:17 +0100 (0:00:00.026)       0:00:00.026 *********** 
ok: [localhost]

TASK [debug] *******************************************************************
Monday 26 June 2023  13:15:17 +0100 (0:00:00.025)       0:00:00.052 *********** 
ok: [localhost] => {
    "my_cmd": "'hello' 'world'\n"
}

TASK [shell] *******************************************************************
Monday 26 June 2023  13:15:17 +0100 (0:00:00.031)       0:00:00.083 *********** 
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "echo 'hello' 'world'\n 'foobar'", "delta": "0:00:00.002549", "end": "2023-06-26 13:15:17.605029", "msg": "non-zero return code", "rc": 127, "start": "2023-06-26 13:15:17.602480", "stderr": "/bin/sh: 2: foobar: not found", "stderr_lines": ["/bin/sh: 2: foobar: not found"], "stdout": "hello world", "stdout_lines": ["hello world"]}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0 
```
You can see that the command fact has a new line in it. but that wont be seen by yaml folding as the templating happens afterwards resulting in the command being broken with a new line in it. 

To avoid this we need to pass a block chomp on the set fact so it doesnt have a trailing new line

```yaml
    - set_fact:
        my_cmd: >-
          'hello'
          'world'
    - debug:
        var: my_cmd
```
Then it works as expected. Can this be reviewed by the community and if `register_runner_cmd` will always be followed by something else please add the chomp block to the set fact like 

```
- name: Construct the runner command without secrets
  set_fact:
    register_runner_cmd: >-
```

**Comment by cdsre** [comment]
I had already written the issue before I noticed that there is already a commit to fix this. https://github.com/riemers/ansible-gitlab-runner/commit/9b28d87e59c13d4b47b2d4446441621edc6d4d2b. Is there any timeline on when this will be released. for now we have pinned back to 1.6.53.

**Comment by riemers** [comment]
v2.0.1 is now coming, try that one

**Comment by cdsre** [comment]
Awesome thanks. Will look out for 2.0.1 in our renovate runner merge request. Appreciate the fast response. I thought I would still submit the issue just for everyone's visibility as issues is the first place I normally come to look for things. I had to dig through the code a bit on this one and by the time I found the issue and fixed and looked at main branch I seen it was already fixed just not release.

Great work on the repo.
